lib/extensions/yaml_load_aliases.rb

```diff
@@ -0,0 +1,20 @@
+module YamlLoadAliases
+  DEFAULT_PERMITTED_CLASSES = [Regexp, Symbol, Date, Time, ActiveSupport::TimeWithZone, ActiveSupport::TimeZone]
+  # Psych 4 aliases load as safe_load.  Some loads happen early, like reading the database.yml so we don't want to load our
+  # constants at that time, such as MiqExpression, Ruport, so we have two sets of permitted classes.
+  def safe_load(*args, **kwargs)
+    super(*args, **kwargs.merge(:aliases => true, :permitted_classes => DEFAULT_PERMITTED_CLASSES))
+  rescue NameError
+    super(*args, **kwargs.merge(:aliases => true, :permitted_classes => DEFAULT_PERMITTED_CLASSES + [MiqExpression, MiqReport, Ruport::Data::Table, Ruport::Data::Record]))
```

Comment:
Note, we're enumerating these classes at method call time to avoid autoloading them at code load time.  This file is loaded very early in the load time.  I suppose we can put this list in an memoized instance method to avoid creating so many arrays.